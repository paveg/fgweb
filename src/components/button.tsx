
type Props = {
  apperance: 'primary' | 'secondary' | 'error'
  onClick?: () => void,
}

const primaryColor = 'border-green-500 bg-green-500 hover:bg-green-600'
const secondaryColor = 'border-yellow-500 bg-yellow-500 hover:bg-yellow-600'
const errorColor = 'border-red-500 bg-red-500 hover:bg-red-600'

export const Button: React.FC<Props> = ({ apperance, onClick, children }): JSX.Element => {
  let color: string
  switch (apperance) {
    case 'primary':
      color = primaryColor
      break
    case "secondary":
      color = secondaryColor
      break
    case 'error':
      color = errorColor
      break
    default:
      color = primaryColor
      break
  }

  return <button
    type="button"
    onClick={onClick}
    className={`${color} border text-white rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}>
    {children}
  </button>
}
