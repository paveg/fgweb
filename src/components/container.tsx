export const Container: React.FC = ({ children }): JSX.Element => {
  return <div className="container mx-auto">
    <main>
      {children}
    </main>
  </div>
}
