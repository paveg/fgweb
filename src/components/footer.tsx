import { IconButton } from './parts/buttons/iconButton'
import TwitterIcon from '@material-ui/icons/Twitter'
import InstagramIcon from '@material-ui/icons/Instagram'

export const Footer: React.VFC = (): JSX.Element => {
  return (
    <footer className="text-gray-600 body-font">
      <div className="container px-5 py-8 mx-auto flex items-center sm:flex-row flex-col">
        <a
          className="flex title-font font-medium items-center md:justify-start justify-center text-gray-900"
          href={'https://fg.vercel.app/'}
        >
          <span className="ml-3 text-xl sm:text-center">Funai Gaming</span>
        </a>
        <p className="text-sm text-gray-500 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-200 sm:py-2 sm:mt-0 mt-4">
          Â© 2021 Funai â€”
          <a
            href="https://twitter.com/paveg_"
            className="text-gray-600 ml-1"
            rel="noopener noreferrer"
            target="_blank"
          >
            @paveg_
          </a>
        </p>
        <span className="inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start">
          <IconButton
            small
            label="Go to Instagram profile page."
            apperance="tertiary"
            IconComponent={InstagramIcon}
            href="https://www.instagram.com/_pavlog/"
            target="_blank"
          />
          <IconButton
            small
            label="Go to Twitter profile page."
            apperance="tertiary"
            IconComponent={TwitterIcon}
            href="https://twitter.com/paveg_"
            target="_blank"
          />
        </span>
      </div>
    </footer>
  )
}
