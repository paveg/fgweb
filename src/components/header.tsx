import * as react from 'react';

type Props = {}

export const Header: React.FC<Props> = ({ children }) => {
    return (
        <nav className="flex items-center justify-between flex-wrap bg-gray-900 p-6">
            <div className="flex items-center flex-no-shrink text-white mr-6">
                <span className="font-bold text-2xl tracking-wide">Funai Gaming Web</span>
            </div>
        </nav>
    )
}
