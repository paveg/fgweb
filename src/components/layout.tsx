import Head from 'next/head'
import { Header } from './header'
import React from 'react'
import { Container } from './container'
import { Footer } from './footer'

type Props = {
  pageTitle: string
}

export const Layout: React.FC<Props> = ({
  pageTitle,
  children,
}): JSX.Element => {
  return (
    <>
      <Head>
        <title>Funai Gaming Web</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Header />
      <Container>
        <h1>{pageTitle}</h1>
        {children}
      </Container>
      <Footer />
    </>
  )
}
