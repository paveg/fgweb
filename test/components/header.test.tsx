import React from 'react'
import { render } from '../testUtils'
import { Header } from '../../src/components/header'

describe('Header', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
