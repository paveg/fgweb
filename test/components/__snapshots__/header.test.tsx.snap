// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center justify-between flex-wrap bg-gray-900 p-6"
  >
    <div
      class="flex items-center flex-no-shrink text-white mr-6"
    >
      <span
        class="font-bold text-2xl tracking-wide"
      >
        Funai Gaming Web
      </span>
    </div>
  </nav>
</DocumentFragment>
`;
